metals/src/main/scala/scala/meta/internal/metals/BloopInstall.scala

```diff
@@ -232,20 +234,28 @@ object BloopInstall {
     }
 
     override def onStdout(buffer: ByteBuffer, closed: Boolean): Unit = {
-      log(closed, buffer)(out => scribe.info(out))
+      log(standardOutput, closed, buffer)(out => scribe.info(out))
     }
 
     override def onStderr(buffer: ByteBuffer, closed: Boolean): Unit = {
-      log(closed, buffer)(out => scribe.error(out))
+      log(errorOutput, closed, buffer)(out => scribe.error(out))
     }
 
-    private def log(closed: Boolean, buffer: ByteBuffer)(
+    private def log(output: StringBuilder, closed: Boolean, buffer: ByteBuffer)(
         fn: String => Unit
     ): Unit = {
       if (!closed) {
-        val text = toPlainString(buffer).trim
-        if (text.nonEmpty) {
-          fn(text)
+        val text = toPlainString(buffer)
+        output.append(text)
+        val lines = output.linesIterator.toList
+        if (output.endsWith("\n")){
+          lines.foreach(fn)
+          output.clear()
+        } else {
+          lines.take(lines.size - 1)
+          val last = lines.last
+          output.clear()
```

Comment:
What happens to the non-last lines? It looks like they get removed here 🤔

Reply:
my bad, forgot foreach - that would come up in maven